irohad/ametsuchi/impl/mutable_storage_impl.cpp

```diff
@@ -64,16 +74,20 @@ namespace iroha {
       };
 
       transaction_->exec("SAVEPOINT savepoint_;");
-      auto result = function(block, *wsv_, top_hash_)
-          and std::all_of(block.transactions.begin(),
-                          block.transactions.end(),
+      // TODO: 14-02-2018 Alexey Chernyshov remove this after relocation to
+      // shared_model https://soramitsu.atlassian.net/browse/IR-881
+      shared_model::crypto::Hash new_hash({top_hash_.begin(), top_hash_.end()});
```

Comment:
Why haven't you changed `top_hash_`'s type as well?

Reply:
#994 is for mutable_storage